cloudinit/net/sysconfig.py

```diff
@@ -634,8 +635,19 @@ def _render_bonding_opts(cls, iface_cfg, iface):
                         bond_value = " ".join(bond_value)
                     bond_opts.append(value_tpl % (bond_value))
                     break
-        if bond_opts:
-            iface_cfg['BONDING_OPTS'] = " ".join(bond_opts)
+        # suse uses the sysconfig support which requires BONDING_MODULE_OPTS
+        # see https://www.kernel.org/doc/Documentation/networking/bonding.txt
+        # 3.1 Configuration with Sysconfig Support
+        if flavor == 'suse':
+            if bond_opts:
+                iface_cfg['BONDING_MODULE_OPTS'] = " ".join(bond_opts)
+        # rhel uses initscript support and thus requires BONDING_OPTS
+        # this is also the old default
+        # see https://www.kernel.org/doc/Documentation/networking/bonding.txt
+        #  3.2 Configuration with Initscripts Support
+        else:
+            if bond_opts:
+                iface_cfg['BONDING_OPTS'] = " ".join(bond_opts)
```

Comment:
```suggestion
        if bond_opts:
            if flavor == 'suse':
                # suse uses the sysconfig support which requires BONDING_MODULE_OPTS
                # see https://www.kernel.org/doc/Documentation/networking/bonding.txt
                # 3.1 Configuration with Sysconfig Support            
                iface_cfg['BONDING_MODULE_OPTS'] = " ".join(bond_opts)
            else:
                # rhel uses initscript support and thus requires BONDING_OPTS
                # this is also the old default
                # see https://www.kernel.org/doc/Documentation/networking/bonding.txt
                #  3.2 Configuration with Initscripts Support
                iface_cfg['BONDING_OPTS'] = " ".join(bond_opts)
```

Reply:
yes that makes more sense this way. I was unsure about the order of the ifs. I'd say we wait for more feedback and do it this way if no one objects.

Reply:
Yes the single `if bond_opts:` check please.